src/algorithms/distance_transform.rs

```diff
@@ -0,0 +1,148 @@
+pub fn distance_transform(room_data: &offline_map::OfflineRoomData) -> LocalCostMatrix {
+  let mut initial_cm = LocalCostMatrix::new();
+
+  for x in 0..ROOM_SIZE {
+    for y in 0..ROOM_SIZE {
+      let position = unsafe { RoomXY::unchecked_new(x as u8, y as u8) };
+      let val = match &room_data.terrain.get_xy(position) {
+        screeps::constants::Terrain::Wall => 0,
+        _ => 255,
+      };
+
+      initial_cm.set(position, val);
+    }
+  }
+
+  distance_transform_from_cost_matrix(initial_cm)
+}
+
+/// Provides a Cost Matrix with values equal to the Chebyshev distance from any position
+/// in the provided initial Cost Matrix with a value set to 0. This allows for calculating
+/// the distance transform from an arbitrary set of positions. Other position values should
+/// be set to 255 to ensure the calculations work correctly.
+pub fn distance_transform_from_cost_matrix(initial_cm: LocalCostMatrix) -> LocalCostMatrix {
+
+  // Copy the initial cost matrix into the output cost matrix
+  let mut cm = initial_cm.clone();
+
+  // Pass 1: Left-to-Right, Top-to-Bottom
+
+  for x in 0..ROOM_SIZE {
+    for y in 0..ROOM_SIZE {
+      let current_position = unsafe { RoomXY::unchecked_new(x as u8, y as u8) };
+
+      // The distance to the closest wall is the minimum of the current position value and
+      // all of its neighbors. However, since we're going RTL:TTB, we can ignore tiles we
+      // know we haven't visited yet: Right, BottomRight, and Bottom. We could include them
+      // and their default max values should get ignored, but why waste the processing cycles?
+      let mut neighbor_values: [u8; 5] = [255; 5];
+
+      if let Some(neighbor_position) = current_position.checked_add_direction(Direction::Top) {
+        let value = cm.get(neighbor_position);
+        neighbor_values[0] = value;
+      }
+
+      if let Some(neighbor_position) = current_position.checked_add_direction(Direction::Left) {
+        let value = cm.get(neighbor_position);
+        neighbor_values[1] = value;
+      }
+
+      if let Some(neighbor_position) = current_position.checked_add_direction(Direction::TopLeft) {
+        let value = cm.get(neighbor_position);
+        neighbor_values[2] = value;
+      }
+
+      if let Some(neighbor_position) = current_position.checked_add_direction(Direction::TopRight) {
+        let value = cm.get(neighbor_position);
+        neighbor_values[3] = value;
+      }
+
+      if let Some(neighbor_position) = current_position.checked_add_direction(Direction::BottomLeft) {
+        let value = cm.get(neighbor_position);
+        neighbor_values[4] = value;
+      }
+
+      let current_value = cm.get(current_position);
+
+      let neighbors_minimum = match neighbor_values.iter().min() {
+        Some(value) => match value {
+          255 => 255,
+          _ => value + 1
+        },
+        None => 255,
+      };
+
+      let min_value = cmp::min(current_value, neighbors_minimum);
```

Comment:
```suggestion
    let min_value = [Direction::Top, Direction::TopLeft, Direction::Left, Direction::BottomLeft].into_iter()
      .filter_map(|dir| current_position.checked_add_direction(dir))
      .map(|position| cm.get(position))
      .min()
      .map(|x| x.saturating_add(1))
      .unwrap_or(255)
      .min(cm.get(current_position));
```

Can do basically the same thing for the reverse direction too.